internal/kafka/internal/services/data_plane_kafka.go

```diff
@@ -379,7 +427,36 @@ func (d *dataPlaneKafkaService) persistKafkaRoutes(kafka *dbapi.KafkaRequest, ka
 	return nil
 }
 
-func buildRoutes(routesInRequest []dbapi.DataPlaneKafkaRouteRequest, kafka *dbapi.KafkaRequest, clusterDNS string) ([]dbapi.DataPlaneKafkaRoute, error) {
+func (d *dataPlaneKafkaService) getManagedKafkaStatus(status *dbapi.DataPlaneKafkaStatus) managedKafkaStatus {
+	for _, c := range status.Conditions {
+		if strings.EqualFold(c.Type, "Ready") {
+			if strings.EqualFold(c.Status, "True") {
+				return statusReady
+			}
+			if strings.EqualFold(c.Status, "Unknown") {
+				return statusUnknown
+			}
+			if strings.EqualFold(c.Reason, "Installing") {
+				return statusInstalling
+			}
+			if strings.EqualFold(c.Reason, "Deleted") {
+				return statusDeleted
+			}
+			if strings.EqualFold(c.Reason, "Error") {
+				return statusError
+			}
+			if strings.EqualFold(c.Reason, "Rejected") {
+				if strings.EqualFold(c.Message, "Cluster has insufficient resources") {
+					return statusRejectedClusterFull
+				}
+				return statusRejected
+			}
```

Comment:
```suggestion
			switch(true) {
				case strings.EqualFold(c.Reason, "Deleted"):
					return statusDeleted
				case strings.EqualFold(c.Reason, "Error"):
					return statusError
				case strings.EqualFold(c.Reason, "Rejected"):
					if strings.EqualFold(c.Message, "Cluster has insufficient resources") {
					return statusRejectedClusterFull
					  }
					return statusRejected
			}
```

Reply:
Thanks for the suggestion. I think I'll keep the `if` statement.